### ISSUE TITLE
Free merchants merchandise update

### ISSUE THREAD

**Issue by I-am-Erk** [CleverRaven/Cataclysm-DDA#51320]
# Is your feature request related to a problem? Please describe.
The free merchants and associated refugees get the short end of the stick for content because they're seen as not worth it by a lot of players. At least some of this is because the store stock is highly random and not reliably helpful.

# Describe the solution you'd like

## 1: Revise the stock that the free merchant vendor carries to mimic "old time dry goods store".
The merchant should sell a wide variety of useful, common components and tools for fabrication, tailoring, and mechanics. In general, if it's annoying to loot and should be common, the merchant should have a copy. Common expendables should be stocked in large quantities. These guys are supposedly a hub for scavengers to bring in goods... they should not have a shortage of this stuff, they should be swimming in it.

The merchant should *not* sell food. Canonically they're usually low on it. The theme should be "raw supplies", like sheets of cloth, thread, toolkits, clay, nails, etc.

## 2: Give the doctor NPC some science gear.
Not every item should be available but the doctor should have some of the common reagents and science tools for sale, allowing you to stock up on these things without raiding research centers.

## 3: add a teamster NPC behind the center with a small garage full of large building supplies.
This fellow should supply the larger things: pipes, lumber in all shapes, logs, sheet metal, rebar, sheets of glass and plastic, spare car parts. Adding a new NPC lets us put them outside so that you can bring up your vehicle and load it easily.

make sure they have guards, they'll be exposed to zombies and things by being outdoors.

Later on, this teamster should be a source for procgen missions to escort caravans and things, in turn adding a story-convenient way to learn about other factions and add them to your map.

# Describe alternatives you've considered
other themes for items would work.

# Additional context
I'd love to do this myself, but let's be honest, I'm spread too thin.

When adding these goods they don't all have to be in the merchant's personal inventory, many could be in item groups stacked on the nearby shelves. The merchants should still have their own stock as well though since that refreshes automatically while the items around them will slowly dwindle.

Some day we should be able to add a flag to itemgroups in mapgen that causes them to respawn over time.

**Comment by eltank** [comment]
Suggestion: the teamster could spawn in the "back bay" after you clear it. The mapgen can be re-written to make it a loading dock area with large garage doors instead of the current bedroom setup. It is called a "bay" after all.

**Comment by I-am-Erk** [comment]
I actually want the refugees to expand into the back area once I can move NPCs. There is already a garage in the back though, and we could put the teamster there.

**Comment by Zeropol** [comment]
Helpless proposition:

Lore-wise feel fit :
- Scrap and bits of metal /electronics / plastic / fabric ( easy to get, but maybe something refugees center has a lot to spare in trade and is always useful. Could be available in large quantities )
- Containers
- Blankets
- Clean water ( the free merchand had to produce a steady supplie of it and once it is done, surplus is inevitable )
- Dirty clothes ( except high end ones like kevlar things )
- Low end weapons / suicide helpers ( knife on a stick, ropes and so on )
- (mis)Information and religious things ( journals, books for fun, talismans )
- Maps of the surroundings

Intencives to players comming by that are not listed above: 
- Thread ( have alternatives yet boring to produce in large quantities )
- Cement ( not everyone use it, but quite rare )
- Basic tools
- Electronics pieces ( power converter, heating element )
- More "rare" raw ressources ( copper, silver, lead, maybe titanium because who need titanium anyway except the player )
- ~~plastic mold~~ All is good now
- Lore snippets

Things that are less welcome :
- High end books ( go explore )
- Bionics obviously
- Bullets and magazines, or at a healthy price
- High end tools

Things I'm in doubt lore and balance wise :
- More common books
- Comestible "byproducts" ( cracklins, cooking oil, sinew, bones )
- Batteries. Disposable batteries may be relevant to trade tho.
- Currency exchange ( implie communication between factions wich would not fit early game )
- Drugs ( amphetamine, heroin and so on.
- Fuel
- Cards ( science, military, industrial )

**Comment by PatrikLundell** [comment]
I like the idea of the free merchants actually having something you'd want to get, rather than the current random junk.

It can be noted that Tacoma seems to have NPCs who are offering services to produce some goods (it was something with wood, such as logs or planks, for instance), and I'd try to avoid competing with those services.

Things I'd be interested in getting:
- Anything I need to build stuff out of in any quantities, so them dismantling stuff and selling the parts would be convenient, since it now takes forever to disassemble most things. This also means I might want to visit them for only a single part rather than spending a day dismantling something to get it.
- Things that are a pain to get in bulk. Clay is definitely a candidate where you may want to pay someone else to do the work for you (especially since there's a [probably now unbalanced] quest that requires you to spend towards 50 hours of active work just to collect it [which I won't do as it's too much work: 3 leather armors is about the limit {14.5 hours of work for PC plus two companions when proficient with all required skills}]).

If it's possible to implement progression, I'd try to make sure only very basic stuff is available early on and in small quantities, with larger volumes and somewhat better things becoming available later. It would cheapen exploration if you just can just go to the merchants to get a complete set of tools on the first day (yes, you need to collect stuff to trade in and they won't expand out back without your help, but still). However, if you reach the middle of the summer (assuming a standard start) and still haven't found a wrench or a hacksaw you should be able to find them there, provided you've put in a sufficient effort in helping them to expand. This would make them more of a backup than a fast track for "normal" tools.

As a tangent, I'd like to see the Free Merchants eventually repair the fence around their place (which ought to help in protecting the new guy in the back).

**Comment by I-am-Erk** [comment]
I think those points are good, we may want to have Tacoma Ranch become the supply point for the more basic "obtained in the wilderness" building materials like clay and logs, and have the Free Merchants supply more developed goods like cloth, rebar, and concrete.

**Comment by Cactoideae** [comment]
I believe the merchant should sell some basic police/swat/army gear in small quantities. The idea is they don't want to use it themselves, because they don't go outside, but they will gladly trade it to you, so YOU can go risk your head and bring them supplies back.

And also, please increase the amount of merch vouchers they are selling or make it infinite. It's annoying when you have enough goods to trade for 750 merch and hire the cool guy, but have to wait for weeks to refresh the inventory.

**Comment by I-am-Erk** [comment]
Remaining to-do: add some more bulk purchase transactions via either the broker or the teamster, influenced by stage of free merchant faction development.

Near future: get those bulk orders special delivered to your faction camp.
